src/middlewared/middlewared/plugins/acme_protocol_/authenticators/shell.py

```diff
@@ -0,0 +1,84 @@
+"""
+The authenticator script is called two times during the certificate generation:
+
+1. The validation record creation which is called in the following way:
+   script set domain validation_name validaton_context timeout
+2. The validation record deletion which is called in following way:
+   script unset domain validation_name validation_context
+
+It is up to script implementation to handle both calls and perform the record creation.
+"""
+import logging
+import subprocess
+
+from middlewared.async_validators import check_path_resides_within_volume
+from middlewared.schema import accepts, Dict, Str, File, Int
+from middlewared.service import CallError, skip_arg, ValidationErrors
+
+from .base import Authenticator
+
+
+logger = logging.getLogger(__name__)
+
+
+class ShellAuthenticator(Authenticator):
+
+    NAME = 'shell'
+    PROPAGATION_DELAY = 60
+
+    SCHEMA = Dict(
+        'shell',
+        File('script', required=True, empty=False, title='Authenticator script'),
+        Str('user', default='nobody', title='Running user', empty=False),
+        Int('timeout', default=60, title='Timeout'),
+        Int('delay', default=60, title='Propagation delay'),
+    )
+
+    def initialize_credentials(self):
+        self.script = self.attributes.get('script')
```

Comment:
Why `get`, can it be absent?

Reply:
I changed this a bit to avoid this case. Thanks for pointing it out!